internal/ircserver/cmd_join.go

```diff
@@ -56,6 +56,7 @@ func (i *IRCServer) cmdJoin(s *Session, reply *Replyctx, msg *irc.Message) {
 			c = &channel{
 				name:  channelname,
 				nicks: make(map[lcNick]*[maxChanMemberStatus]bool),
+				key:   key,
```

Comment:
This change surprises me: when a channel does not exist yet, I think it should be created without a key, regardless of whether the user tried to use a key to join the channel.

Whether a channel has a key or not is determined by the 'k' channel mode, so setting the struct field “key” without adjusting the modes is inconsistent (but, again, I think we should probably just remove this line).

Reply:
Missed this, fixed.